keps/3589-manage-jobs-selectively/README.md

```diff
@@ -0,0 +1,149 @@
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+- [Design Details](#design-details)
+  - [API](#api)
+  - [Implementation](#implementation)
+  - [Test Plan](#test-plan)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+This KEP adds an optional field to the Kueue `Configuration` struct that enables
+the effect of `manageJobsWithoutQueueName` to be controlled on a per-namespace
+basis for all Kueue integrations.
+
+-->
+
+## Motivation
+
+Kueue already allows per-namespace control of `manageJobsWithoutQueueName`
+for the Pod integration via `integrations.podOptions.namespaceSelector`.
+The effect of `manageJobsWithoutQueueName` on the `Deployment` and `StatefulSet`
+integrations in Kueue 0.9 are also indirectly controlled by the same `namespaceSelector`
+because these integrations are built on top of the `Pod` integration.  Just like the
+`Pod` integration, Kueue requires some mechanism for modulating `manageJobsWithoutQueueName`
+for these fundamental types for system namespaces to avoid interrupting normal cluster operations.
+
+No other integrations beyond these three consider the `namespaceSelector` when deciding whether or not to
+suspend jobs that do not have a queue name.  This results in an irregular API and makes it
+harder to configure Kueue such that quotas will be enforced (cannot be bypassed by users).
+The irregularity between `batchv1/Job` and `Deployment` is likely to be especially confusing
+to users and cluster admins who would view both of these types as "built-in" to Kubernetes and
+would expect them to be configured in the same way.
+
+### Goals
+
+Provide a simple and uniform API to enable cluster admins to configure Kueue
+such that quotas will be enforced for specific namespaces.
+
+### Non-Goals
+
+Consider any scope for filtering `manageJobsWithoutQueueName` that is not based on namespaces.
+
+## Proposal
+
+We add a `manageAllJobsNamespaceSelector` of type `*metav1.LabelSelector`
+to the top level of the Kueue `Configuration` struct (same level as `manageJobsWithoutQueueName`).
+(The name of `manageJobsWithoutQueueNameNamespaceSelector` reads poorly, so I propose something else)
+
+We use this new configuration for all integrations in the same way that the current
+`namespaceSelector` is used by the Pod integration.
+
+We migrate away from using `podOptions.namespaceSelector` over the course of several releases.
+Ideally we remove it from the `Configuration` struct as part of going to the `v1` API level.
+
+### User Stories (Optional)
+
+#### Story 1
+
+Cluster admins will be able to use `manageAllJobsNamespaceSelector` to easily configure Kueue
```

Comment:
Reading this, I don't quite understand the interaction of ManageJobsWithoutQueueName and ManageAllJobsNamespaceSelector. Could you give an example of how you would configure this and how it would work?

I'm just not entirely sure what someone would put for a label selector in this case.

Reply:
I expanded the proposal and user story to address this feedback.